packages/sync/src/offline/OfflineQueue.ts

```diff
@@ -23,79 +27,91 @@ export interface OfflineQueueOptions {
  * - persisting operation queue in provided storage
  * - updating client IDs with server IDs (explained below)
  */
-export class OfflineQueue extends OperationQueue {
+export class OfflineQueue {
+  public queue: OperationQueueEntry[] = [];
   private readonly storage?: PersistentStore<PersistedData>;
   private readonly storageKey?: string;
   private readonly listener?: OfflineQueueListener;
   private readonly state?: ObjectState;
 
-  constructor(options: OfflineQueueOptions) {
-    super(options);
-
-    const { storage, storageKey, listener, conflictStateProvider } = options;
+  constructor(options: OfflineLinkOptions) {
+    this.storage = options.storage;
+    this.storageKey = options.storageKey;
+    this.listener = options.listener;
+    this.state = options.conflictStateProvider;
+  }
 
-    this.storage = storage;
-    this.storageKey = storageKey;
-    this.listener = listener;
-    this.state = conflictStateProvider;
+  public enqueue(operation: Operation, forward: NextLink) {
+    const operationEntry = new OperationQueueEntry(operation, forward);
+    this.enqueueEntry(operationEntry);
+    return new Observable((observer) => {
+      operationEntry.observer = observer;
+      return () => ({ isOffline: true });
+    });
   }
 
-  /**
-   * Returns list of operations that can be forwarded - i.e. they have not
-   * been forwarded yet and do not have client ID.
-   */
-  public toBeForwarded() {
-    return this.queue.filter(op => !op.subscription && !op.hasClientId());
+  public async forwardOperations() {
+    for (const op of this.queue) {
+      // FIXME block operations till result is back (completed)
+      await new Promise((resolve, reject) => {
+        op.forward(op.operation).subscribe({
+          next: (result: FetchResult) => {
+            this.queue = this.queue.filter(e => e !== op);
```

Comment:
This is hard to read. I would love to have next and error handled by separate methods for better readability.